src/classes/ALLO_AllocationsReviewService.cls

```diff
@@ -0,0 +1,232 @@
+     * @description Determines if multi-currency is enabled for the org
+     */
+    private Boolean isMultiCurrencyEnabled {
+        get {
+            if (isMultiCurrencyEnabled == null) {
+                isMultiCurrencyEnabled = UserInfo.isMultiCurrencyOrganization();
+            }
+            return isMultiCurrencyEnabled;
+        }
+        set;
+    }
+
+    /**
+     * @description Allocations Wrapper Allocations Work
+     */
+    private ALLO_AllocationsWrapper allocationsWrapper {
+        private get {
+            if (allocationsWrapper == null) {
+                allocationsWrapper = new ALLO_AllocationsWrapper();
+            }
+            return allocationsWrapper;
+        }
+        private set;
+    }
+
+    /**
+     * @description Provides an allocations wrapper to the class for processing
+     * @param allocationsWrapper Allocation Wrapper with DML to be processed
+     * @return Class Instance
+     */
+    public ALLO_AllocationsReviewService withAllocationsWrapper(ALLO_AllocationsWrapper allocationsWrapper) {
+        this.allocationsWrapper = allocationsWrapper;
+        return this;
+    }
+
+    /**
+     * @description Evaluates the data in the allocationsWrapper and returns a set consisting of any Opportunity Ids
+     *              that require processing
+     * @return Set<Id> Set of Opportunity Ids that requires further processing
+     */
+    public Set<Id> evaluateAllocations() {
+        Set<Id> opportunityIds = new Set<Id>();
+
+        for (Opportunity opp: allocationsWrapper.oppsById.values()) {
+
+            if (checkOpportunity(opp)) {
+                opportunityIds.add(opp.Id);
+            }
+        }
+
+        return opportunityIds;
+    }
+
+    /**
+     * @description Checks Opportunity and Related Allocations Information (including Payment Allocations and
+     *              Defaults if enabled) and returns true if it needs processing
+     * @param opp Opportunity to review
+     * @return Boolean true if it requires processing
+     */
+    @testVisible private Boolean checkOpportunity(Opportunity opp) {
+        Decimal amount = ALLO_AllocationsUtil.getAmountFor(opp);
+        List<Allocation__c> oppAllocs = allocationsWrapper.getOpportunityAllocationsFor(opp);
+        List<Allocation__c> campaignAllocs = allocationsWrapper.getCampaignAllocationsFor(opp);
+        List<Allocation__c> recurringDonationAllocs = allocationsWrapper.getRecurringDonationAllocationsFor(opp);
+
+        List<Allocation__c> parentAllocs = (!campaignAllocs.isEmpty()) ? campaignAllocs : recurringDonationAllocs;
+
+        if (isRequiresReview(opp, oppAllocs, parentAllocs)) {
+            // As soon as we find at least one reason to review we stop looking
+            return true;
+        }
+        // Payment Allocations are reviewed if Payment Allocations is enabled
+        if (ALLO_AllocationsSettings.getSettings().isPaymentAllocationsEnabled()) {
+            if (opp.npe01__OppPayment__r != null) {
```

Comment:
```suggestion
        if (!ALLO_AllocationsSettings.getSettings().isPaymentAllocationsEnabled() || opp.npe01__OppPayment__r != null) {
            return false;
        }
```

Reply:
Need to look at the block to make sure that does not alter the flow, but it doesn't I will make the change

Reply:
@force2b - I keep looking at this one, and the change does not appear to do what I need it to.  I need to enter the block and do additional Payment Allocations checks for each payment, if Payment Allocations is Enabled, and if the Opportunity has Payments. If Payment Allocs are required, and the opp has allocs, i have to then review each payments allocations for changes.

Reply:
Disregrard.  It took me a bit, but I think I figured out what you meant.